families/battleship/Dockerfile

```diff
@@ -12,38 +12,36 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# docker build -f families/battleship/Dockerfile -t sawtooth-battleship-tp-local .
-
-# -------------===  build ===-------------
+# docker build -f families/battleship/Dockerfile -t sawtooth-battleship-local .
 
 FROM ubuntu:bionic
 
 RUN apt-get update \
- && apt-get install gnupg -y
-
-RUN echo "deb http://repo.sawtooth.me/ubuntu/ci bionic universe" >> /etc/apt/sources.list \
- && (apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 8AA7AF1F1091A5FD \
- || apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 8AA7AF1F1091A5FD) \
- && apt-get update
+ && apt-get install -y \
+ curl \
+ gcc \
+ libssl-dev \
+ libzmq3-dev \
+ pkg-config \
+ unzip
 
-RUN apt-get install -y -q \
-    git \
-    python3 \
-    python3-stdeb
+# For Building Protobufs
+RUN curl -OLsS https://github.com/google/protobuf/releases/download/v3.5.1/protoc-3.5.1-linux-x86_64.zip \
+ && unzip protoc-3.5.1-linux-x86_64.zip -d protoc3 \
+ && rm protoc-3.5.1-linux-x86_64.zip
 
-RUN apt-get install -y -q \
-    python3-grpcio \
-    python3-grpcio-tools \
-    python3-protobuf
 
-RUN apt-get install -y -q \
-    python3-colorlog \
-    python3-toml \
-    python3-yaml \
-    python3-zmq
+ENV PATH=$PATH:/project/sawtooth-core/bin:/protoc3/bin:/root/.cargo/bin \
+    CARGO_INCREMENTAL=0
 
-RUN mkdir -p /var/log/sawtooth
-
-ENV PATH=$PATH:/project/sawtooth-core/bin
+RUN curl https://sh.rustup.rs -sSf > /usr/bin/rustup-init \
+ && chmod +x /usr/bin/rustup-init \
+ && rustup-init -y
 
 WORKDIR /project/sawtooth-core
+
+CMD cd families/battleship \
+ && echo "\033[0;32m--- Building battleship ---\n\033[0m" \
+ && cargo build --release \
+ && cp ./target/release/processor /project/sawtooth-core/bin/battleship-tp \
```

Comment:
I think `./target/release/processor` should be `./target/release/battleship`.

Reply:
I've got the binaries under `src/bin/` as `processor` and `cli`, so unless we change those, I think this needs to be `./target/release/processor`.